### PR TITLE
jsPDF export - add pdf_exporter, tests

### PR DESCRIPTION
**Pull request by KuznetsovVN**

### CODE REVIEW

File: js/exporter/jspdf/export.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                        const pdfCell = {
+                            content: cellData.value,
+                            styles: {}
+                        };
+
+                        this.assignCellStyle(pdfCell, gridCell, columns[cellIndex], cellStyle);
+
+                        rowType = rowType || gridCell.rowType;
+                        if(rowType === 'header') {
+                            if(pdfCell.content !== '') { row.push(pdfCell); }
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export.js]: else ?
Or new entry into TODO check list?
Column headers will be incorrectly shifted after the first empty line.

File: js/exporter/jspdf/export.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+            k = 12;
+        } else if(unitType === 'em') {
+            k = 12;
+        } else if(unitType === 'ex') {
+            k = 6;
+        }
+        return k;
+    },
+
+    convertUnitToPoints: function(value, unitType) {
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export.js]: There are these calls:
1. autoTableOptions.tableWidth = this.convertUnitToPoints(tableWidth, 'px');
2. columnStyles[i].cellWidth = this.convertUnitToPoints(columnWidth, 'px');

Let's reduce it to 'convertPixelsToPoints' and keep 'px' only conversion:

    k = 96 / 72; // magic

File: js/exporter/jspdf/export_data_grid.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    if(!(isDefined(options) && isObject(options))) {
+        throw Error('The "exportDataGrid" method requires a configuration object.');
+    }
+    if(!(isDefined(options.component) && isObject(options.component) && options.component instanceof DataGrid)) {
+        throw Error('The "component" field must contain a DataGrid instance.');
+    }
+    if(!isDefined(options.selectedRowsOnly)) {
+        options.selectedRowsOnly = false;
+    }
+    if(!isDefined(options.autoFilterEnabled)) {
```
Comment (IgnatovDan) [inline js/exporter/jspdf/export_data_grid.js]: autoFilterEnabled for pdf? Remove?

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/exporter/jspdf/export.js (hunk 1 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export.js]: Yes, it is wrong to check the contents of the header cell. I will remove this condition.
In the future, we will implement the header logic for BAND here.

File: js/exporter/jspdf/export.js (hunk 2 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export.js]: It is a scale parameter for conversion from 'px' to 'pt'.
72 is a points per inch
96 is a pixels per inch
```
px = 96 / 72 * pt
px = 4 / 3 * pt
pt = 3 / 4 * px
```
I will reduce these methods to the 'convertPixelsToPoints' method.

File: js/exporter/jspdf/export_data_grid.js (hunk 3 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/export_data_grid.js]: Yes, this option is not needed here.
Removed

**Review by IgnatovDan — APPROVED** [review]

**Pull request merged**